src/kv/kv_types.h

```diff
@@ -528,6 +528,30 @@ namespace kv
     virtual void swap(AbstractMap* map) = 0;
   };
 
+  class Tx;
+
+  class AbstractExecutionWrapper
+  {
+  public:
+    virtual ~AbstractExecutionWrapper() = default;
+    virtual kv::ApplySuccess execute() = 0;
+    virtual kv::ConsensusHookPtrs& get_hooks() = 0;
+    virtual const std::vector<uint8_t>& get_entry() = 0;
+    virtual kv::Term get_term() = 0;
+    virtual kv::Version get_index()
```

Comment:
Why do we have stubs in an Abstract class? The test stubs seem to reimplement those as well, so it seems like they could be pure virtual, unless I've missed something?

Reply:
yes good point, I am making it pure virtual